fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -233,17 +253,21 @@ def predict(self, X):
             the result will be a scalar. Otherwise the result will be an
         :rtype: Scalar or array
         """
+        if self.best_result is None:
+            raise NotFittedException(_NO_PREDICT_BEFORE_FIT)
```

Comment:
should this be self._best_result? does the property magic take care of this somehow?

Reply:
Yes. Magic in the property makes this work. However, by only having the 'getter' property public, users are dissuaded from fiddling with the results.